### DISCUSSION TITLE
Building behind China's firewall issue?

### DISCUSSION THREAD

**Discussion by olivier2** [KRTirtho/spotube#1262] [Q&A]
Continuing build support thread started in #88 with @wuyuansheng1982 . [Last error](https://github.com/KRTirtho/spotube/issues/88#issuecomment-1962776531) copy-pasted here:

```
PS C:\repos\github\spotube> flutter run -d windows
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Launching lib\main.dart on Windows in debug mode...
CMake Error at flutter/ephemeral/.plugin_symlinks/media_kit_libs_windows_audio/windows/CMakeLists.txt:40 (message):
 
  C:/Repos/Github/spotube/build/windows/x64/mpv-dev-x86_64-20230924-git-652a1dd.7z
  Integrity check failed, please try to re-build project again.
Call Stack (most recent call first):
  flutter/ephemeral/.plugin_symlinks/media_kit_libs_windows_audio/windows/CMakeLists.txt:76 (download_and_verify)

```

**Comment by olivier2** [comment]
@wuyuansheng1982, could you try deleting:
`C:/Repos/Github/spotube/build/windows/x64/mpv-dev-x86_64-20230924-git-652a1dd.7z`

and try to run again:
`flutter run -d windows`

Hopefully that will try to download the file again.

Other option is to try and download the 7z archive directly from [here](https://github.com/media-kit/libmpv-win32-audio-build/releases) and place the file at the right location : C:/Repos/Github/spotube/build/windows/x64/mpv-dev-x86_64-20230924-git-652a1dd.7z ?

**Reply by wuyuansheng1982** [reply]
I downloaded the file as suggested, now it gives me a different error message.

PS C:\repos\github\spotube> flutter run -d windows
Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
Launching lib\main.dart on Windows in debug mode...
Package 'Microsoft.Windows.CppWinRT 2.0.210806.1' is not found in the following primary source(s): 'C:\Users\Eddy Wu\.nuget\packages\,C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'. Please verify all your online package sources are available (OR) package id, version are specified correctly.
CMake Error at flutter/ephemeral/.plugin_symlinks/permission_handler_windows/windows/CMakeLists.txt:29 (message):
  Failed to install nuget package Microsoft.Windows.CppWinRT.2.0.210806.1


Building Windows application...                                  1,080ms
Error: Unable to generate build files

Searched for answers, not quite sure how to proceed

**Comment by olivier2** [comment]
Have a look at this [thread](https://github.com/Baseflow/flutter-permission-handler/issues/819); specifically this answer [here](https://github.com/Baseflow/flutter-permission-handler/issues/819#issuecomment-1673271795) where a couple of verifications are suggested. If it stills fail, share the result of `flutter build windows -v`

**Comment by wuyuansheng1982** [comment]
great, the answer helped me go past the nuget error. 

Now I am getting a build error. 'flutter run -d windows', the error message is a bit too hard to look at. I tried to build in VS studio, think it's the same problem

![image](https://github.com/KRTirtho/spotube/assets/240229/97f0723d-3ad5-4936-bce4-8509afe0178f)

**Comment by olivier2** [comment]
All I can figure out is that the cmake command failed, but can't tell from the screenshot what the actual error is. Maybe a log from the command line would show the details?

**Comment by wuyuansheng1982** [comment]
about the same, https://pastebin.com/tDNL6GTT

**Comment by olivierIllogika** [comment]
Let's have a look at the output of `flutter doctor -v`

If it's all green, I suggest starting from scratch (making sure you still have the .env file) with:
```
flutter clean
flutter pub get
dart run build_runner build --delete-conflicting-outputs --enable-experiment=records,patterns
flutter run -v -d windows
```

**Reply by wuyuansheng1982** [reply]
flutter doctor -v
        https://pastebin.com/b4RcJVDK
flutter run -v -d windows, no issues until the cmake error.
        https://pastebin.com/Afnp6cVV

**Comment by olivier2** [comment]
Looks to me like another issue while downloading packages in `flutter pub get` step. The package `metadata_god` seems corrupt or incomplete.

Try deleting the `metadata_god****` folder in your package cache. It should be in:
`%APPDATA%\Pub\Cache\hosted`

And then run `flutter clean` and `flutter pub get` again.

**Reply by KRTirtho** [reply]
That's because currently, pub.dev only allows upto 100MB of asset inside a package. So, before the first build/debug-build the required shared-objects, dlls, xcframeworks (it's the elephant in the room) are fetched from GitHub release. Any package that uses external native libraries (not native source code), have to take this approach :')

**Reply by KRTirtho** [reply]
@wuyuansheng1982  This is the [CMake file](https://github.com/KRTirtho/metadata_god/blob/main/packages/metadata_god/windows/CMakeLists.txt) that downloads the assets if not already downloaded.
You can just manually download those [assets for the version](https://github.com/KRTirtho/metadata_god/releases/tag/v0.5.2) you're using and place them in that directory

**Reply by wuyuansheng1982** [reply]
I got to the point where exes have been installed to c:\program files\metadata-god, tried running, doesn't work.

There are still errors in the build log. 

https://pastebin.com/jgyxRMnE
